pkg/agent/controller/bgp/controller.go

```diff
@@ -991,3 +991,40 @@ func (c *Controller) GetBGPPeerStatus(ctx context.Context) ([]bgp.PeerStatus, er
 	}
 	return peers, nil
 }
+
+// GetBGPRoutes returns the advertised BGP routes.
+func (c *Controller) GetBGPRoutes(ctx context.Context, ipv4Routes, ipv6Routes bool) (sets.Set[string], error) {
+	bgpRoutes := make(sets.Set[string])
+	bgpRoutesAdvertised := c.getBgpRoutesAdvertised()
+	if bgpRoutesAdvertised == nil {
+		return nil, ErrBGPPolicyNotFound
+	}
+	if ipv4Routes && ipv6Routes { // return all advertised routes
+		for route := range bgpRoutesAdvertised {
+			bgpRoutes.Insert(route.Prefix)
```

Comment:
I am wondering what's the order of the output if it's a dual stack? 
Is it ok to be in a random order for IPv4 and IPv6?  @antoninbas

Reply:
changed to insert ipv4 routes first then ipv6 routes.